auth0/src/main/java/com/auth0/android/authentication/storage/CryptoUtil.java

```diff
@@ -145,6 +146,22 @@ public CryptoUtil(@NonNull Context context, @NonNull Storage storage, @NonNull S
         }
     }
 
+    private KeyStore.PrivateKeyEntry getKeyEntryCompat(KeyStore keyStore)
+            throws KeyStoreException, NoSuchAlgorithmException, UnrecoverableEntryException {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            //Following code is for API 28+
+            PrivateKey privateKey = (PrivateKey) keyStore.getKey(KEY_ALIAS, null);
+            Certificate certificate = keyStore.getCertificate(KEY_ALIAS);
+
+            if (privateKey == null) {
+                return (KeyStore.PrivateKeyEntry) keyStore.getEntry(KEY_ALIAS, null);
```

Comment:
this line will have the same value as `privateKey`. Seems like this will always return null, am I correct?

Reply:
When the `privateKey` is null, the constructor can't be used to create a PrivateKeyEntry object and it throws a NullPointerException. But the method here returns a PrivateKeyEntry object that contains a null privateKey, and an empty certificate chain without throwing any exceptions or warnings.